### ISSUE TITLE
Refactor get_final_element so it parses a MultiAddr and returns Result

### ISSUE THREAD

**Issue by adrianbenavides** [build-trust/ockam#3554]
Function: https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/util/mod.rs#L456

## Current

The `get_final_element` function parses a string that could be either a valid `node` type `MultiAddr` or just contain the name of a node (i.e. either `/node/<name>` or `<name>`). 

This function must return the node name.

Currently this is naively implemented, splitting the input string by `/` and extracting the last item.

## Desired

We want to refactor this method so it parses the input string as a `MultiAddr` like it's done [here](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/forwarder/create.rs#L58) for example and it returns a `Result<String>` instead of calling the `std::process::exit` function.

A possible solution could look like this (note that the function and the argument have been renamed):

```rust
///  Get node name from a string with format `/node/<name>` or `<name>`  
pub fn extract_node_name(input: &str) -> anyhow::Result<&str> {  
    // if input has "/", we process it as a MultiAddr  
    if input.contains("/node/") {  
        let err = anyhow!("invalid node address protocol");  
        let maddr = MultiAddr::from_str(input)?;  
        if let Some(p) = maddr.iter().next() {  
            if p.code() == Node::CODE {  
                let name = p.cast::<Node>().ok_or_else(|| err)?;  
                Ok(name.to_string().as_str())  
            } else {  
                Err(err)  
            }  
        } else {  
            Err(err)  
        }  
    }  
    // else we have a simple string, so we return it as it is  
    else {  
        Ok(input)  
    }  
}
```

We would also expect to have some unit tests to make sure it's working with both expected formats. You can add them at the end of [this file](https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_command/src/util/mod.rs).

---

_We love helping new contributors!_
_If you have questions or need help as you work on your first Ockam contribution, please leave a comment on [this discussion](https://github.com/ockam-network/ockam/discussions/1081)._
_If you're looking for other issues to contribute to, checkout [this discussion](https://github.com/build-trust/ockam/discussions/3206) and labels -  https://github.com/build-trust/ockam/labels/good%20first%20issue or https://github.com/build-trust/ockam/labels/help%20wanted_

**Comment by michealkeines** [comment]
Hi @adrianbenavides i would like to work on this.

**Comment by adrianbenavides** [comment]
All yours! Let me know if you have any question :+1:

**Comment by michealkeines** [comment]
Hi @adrianbenavides I have added the changes in [PR](https://github.com/build-trust/ockam/pull/3608), Please review it and let me if there are any changes needed.

Thank you.

**Comment by elpiel** [comment]
Is this resolved after the #3608 PR was merged?

**Comment by adrianbenavides** [comment]
Yes, that's right! Thanks for pointing that out @elpiel
